tests/graph/test_hydrate.py

```diff
@@ -52,6 +53,31 @@ class Tree(BaseModel):
     branches: list[Branch]
 
 
+class Caterpillar(BaseModel):
+    home: Leaf | None
+
+
+@pytest.mark.parametrize(
+    ("model", "attribute", "expected"),
+    [
+        (Tree, "branch", Branch),
+        (Branch, "leaves", Leaf),
+        (Leaf, "color", "cannot hydrate paths with non base models"),
+        (Leaf, "veins", "cannot hydrate paths with non base models"),
+        (Caterpillar, "home", Leaf),
+    ],
+)
+def test__get_base_model_from_field(
```

Comment:
```suggestion
def test_get_base_model_from_field(
```

i know, it's technically correct with `__`, it just looks irritating to me